knot.x/modules/scenarios/src/main/java/io/knotx/scenarios/scenario0/action/PaymentsActionFactory.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.scenarios.scenario0.action;
+
+import static io.knotx.scenarios.scenario0.utils.ProvidersProvider.calculateProviders;
+
+import org.apache.commons.lang3.StringUtils;
+
+import io.knotx.fragments.handler.api.Action;
+import io.knotx.fragments.handler.api.ActionFactory;
+import io.knotx.fragments.handler.api.domain.FragmentResult;
+import io.reactivex.Single;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+
+public class PaymentsActionFactory implements ActionFactory {
```

Comment:
This action is stateless so the factory can be marked with `io.knotx.fragments.handler.api.Cacheable` to prevent Action creation every time it is invoked. See `io.knotx.fragments.handler.action.InMemoryCacheActionFactory` as the example.

Reply:
You are right. Changed.